pages/docs/plugins/grouparoo-postgres.mdx

```diff
@@ -0,0 +1,205 @@
+### Query Source
+
+A Query Source is a more flexible way to build properties. With a Query Source, you can add custom SQL commands to your Properties, which could pull data from one or more tables in your database.
+
+The unique thing about the way a Query Source works is that there aren't any unique options for the Source itself. But the Schedule and Properties look a little different.
+
+#### Query Source Schedule
+
+Like a [Table Source Schedule](#table-source-schedule), a Query Source Schedule is included with the generated config file, commented. It has a couple unique options:
+
+- `query`: A SQL query to return that tells Grouparoo which Profiles to check each time the interval occurs.
+- `propertyId`: The `id` of the Grouparoo Property whose data is returned by `options.query`.
+
+Here's an example:
+
+```js
+{
+  options: {
+    query: "SELECT id FROM users WHERE updated_at >= (NOW() - INTERVAL '2 day')",
+    propertyId: "userId"
+  }
+}
+```
+
+#### Query Source Properties
+
+Query Sources are a little simpler when it comes to Properties. You can generate a Property using the CLI:
+
+    $ grouparoo generate postgres:query:property lifetime_value --parent purchases
+
+This will drop a file at `config/properties/lifetime_value.js`. Edit this file to match your desired configuration.
+
+There is one unique option for Query Source Properties:
+
+- `query`: The query to extract the Property. You can use [mustache variables](https://github.com/janl/mustache.js#variables) to represent the keys of other Properties in the system.
+
+Here's an example that sums the values in the `price` column for rows in which the `user_id` column's value matches the value of the Grouparoo Profile's `userId` field (i.e. `userId` is the `id` for the Property in Grouparoo):
+
+```js
+{
+  options: {
+    query: "SELECT SUM(price) from purchases where user_id = {{userId}}";
+  }
+}
+```
+
+## Creating a Postgres Destination
+
+Postgres is one of the few integrations that can be both a Source and a Destination. You can use Grouparoo to export data to a Postgres database. Like a Source, a Destination requires that you've already configured an App to describe the connection details to you Postgres database.
+
+Once your App exists, you can generate the config file for the Destination:
+
+    $ grouparoo generate postgres:destination my_exports
+
+This would generate a config file at `config/destinations/my_export.js` in your Grouparoo project. Edit this file to match your desired configuration.
+
+The Postgres Destination has a few unique options:
+
+- `table`: The name of the table which Grouparoo which use to export **Profiles**.
+- `primaryKey`: The primary key for the specified table.
+- `groupsTable`: Grouparoo also exports group memberships. This is the table used to export Group data.
+- `groupForeignKey`: The foreign key that the groups table uses to reference the profiles table.
+- `groupColumnName`: The column name used to store the Group name values.
+
```

Comment:
The Postgres Destination will make 2 tables - one for Profiles, and another normalized table for Group membership.